plugins/github-deployments/src/components/GithubDeploymentsCard.test.tsx

```diff
@@ -199,5 +244,120 @@ describe('github-deployments', () => {
+          'backstage.io/source-location': 'github:my-github-1.com',
+        };
+      });
+
+      it('should fetch data using custom api', async () => {
+        worker.use(
+          GRAPHQL_CUSTOM_API_1.query('deployments', (_, res, ctx) =>
+            res(ctx.data(responseStub)),
+          ),
+        );
+
+        await assertFetchedData();
+        expect.assertions(7);
+      });
+    });
+
+    describe('entity with managed by location', () => {
+      beforeEach(() => {
+        entity = entityStub;
+        entity.entity.metadata.annotations = {
+          'github.com/project-slug': 'org/repo',
+          'backstage.io/managed-by-location': 'github:my-github-2.com',
+        };
+      });
+
+      it('should fetch data using custom api', async () => {
+        worker.use(
+          GRAPHQL_CUSTOM_API_2.query('deployments', (_, res, ctx) =>
+            res(ctx.data(responseStub)),
+          ),
+        );
+
+        await assertFetchedData();
+        expect.assertions(7);
+      });
+    });
+
+    describe('entity with location without baseApiURL', () => {
+      beforeEach(() => {
+        entity = entityStub;
+        entity.entity.metadata.annotations = {
+          'github.com/project-slug': 'org/repo',
+          'backstage.io/managed-by-location': 'github:my-github-3.com',
+        };
+      });
+
+      it('shows no apiBaseUrl error', async () => {
+        const rendered = await renderInTestApp(
+          <ApiProvider apis={apis}>
+            <GithubDeploymentsCard />
+          </ApiProvider>,
+        );
+
+        expect(
+          await rendered.findByText(
+            'Warning: No apiBaseUrl available for host github:my-github-3.com, please check your integrations config',
+          ),
+        ).toBeInTheDocument();
+      });
+    });
+
+    describe('entity with location without matching host', () => {
+      beforeEach(() => {
+        entity = entityStub;
+        entity.entity.metadata.annotations = {
+          'github.com/project-slug': 'org/repo',
+          'backstage.io/managed-by-location':
+            'github:my-github-unknown.com/org/repo',
+        };
+      });
+
+      it('shows no matching host error', async () => {
+        const rendered = await renderInTestApp(
+          <ApiProvider apis={apis}>
+            <GithubDeploymentsCard />
+          </ApiProvider>,
+        );
+
+        expect(
+          await rendered.findByText(
+            'Warning: No matching GitHub integration configuration for host github:my-github-unknown.com/org/repo, please check your integrations config',
+          ),
+        ).toBeInTheDocument();
+      });
+    });
+
+    describe('entity with non url location', () => {
+      beforeEach(() => {
+        entity = entityStub;
+        entity.entity.metadata.annotations = {
+          'github.com/project-slug': 'org/repo',
+          'backstage.io/source-location':
+            'some-other-managed-location:my-favourite-location/org/repo',
```

Comment:
Do we have tests for the `url:` protocol in here somewhere?

Reply:
You mean a test asserting url isn't used and default github host is?

Reply:
Added a separate test for `url:`.